destination/destination_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"testing"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/golang/mock/gomock"
+	"github.com/matryer/is"
+
+	"github.com/conduitio-labs/conduit-connector-mongo/config"
+	"github.com/conduitio-labs/conduit-connector-mongo/destination/mock"
+)
+
+func TestDestination_Configure_success(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	d := Destination{}
+	err := d.Configure(context.Background(), map[string]string{
+		config.KeyURI:        "mongodb://localhost:27017",
+		config.KeyDB:         "test",
+		config.KeyCollection: "users",
+	})
+
+	is.NoErr(err)
+
+	is.Equal(d.config, config.Config{
+		URI: &url.URL{
+			Scheme: "mongodb",
+			Host:   "localhost:27017",
+		},
+		DB:         "test",
+		Collection: "users",
+	})
+}
+
+func TestDestination_Configure_mechanismFailure(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	d := Destination{}
+	err := d.Configure(context.Background(), map[string]string{
+		config.KeyURI:           "mongodb://localhost:27017",
+		config.KeyDB:            "test",
+		config.KeyCollection:    "users",
+		config.KeyAuthMechanism: "not existing mechanism",
+	})
+
+	is.Equal(err.Error(), "parse config: invalid auth mechanism \"NOT EXISTING MECHANISM\"")
+}
+
+func TestDestination_Configure_structValidateFailure(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	d := Destination{}
+	err := d.Configure(context.Background(), map[string]string{
+		config.KeyURI:        "mong\\'odb://localhost:27017",
+		config.KeyDB:         "test",
+		config.KeyCollection: "users",
+	})
+
+	is.True(err != nil)
+}
+
+func TestDestination_Write_success(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+
+	it := mock.NewMockWriter(ctrl)
+	it.EXPECT().Write(ctx, sdk.Record{}).Return(nil)
+
+	d := Destination{
+		writer: it,
+	}
+
+	lines, err := d.Write(ctx, []sdk.Record{{}})
```

Comment:
lines = count?

Reply:
Correct.
https://github.com/conduitio-labs/conduit-connector-mongo/blob/destination/destination/destination.go#L155
Do you mean I should rename it?

Reply:
I think that would be nice, but not a must.:)

Reply:
Done 👍